### DISCUSSION TITLE
Give ability to modify changeOrigin proxy parameter

### DISCUSSION THREAD

**Discussion by lechinoix** [vercel/next.js#47895] [Ideas]
### Goals

Being able to redirect to a server preserving original Host.

### Background

I have a Shopify Theme frontend that I currently migrate to Next.JS. I use the rewrite/fallback feature to di this. To preserve the way shopify works today, I need to keep telling him that it is served on my domain (barooders.com), but really it is my next application that is on barooders.com. To do the redirect, I therefore need to bypass the DNS, use the IP address of shopify and send him barooders.com as a Host header. This does not work because the changeOrigin is hardcoded to true in the proxy configuration.

### Proposal

I propose to add an experimental config as there is for the proxyTimeout :

https://github.com/vercel/next.js/pull/47896/commits/4e3a0ef58663cff41b428b905eef589fba213ecb

**Comment by ghost** [comment]
I used a seperate proxy to change the host header and it solved this issue.
